File: backend/app/schemas/subject.py
from typing import Optional
from pydantic import BaseModel
from datetime import datetime

# Shared properties
class SubjectBase(BaseModel):
    subject_name: str
    subject_code: str
    description: Optional[str] = None
    major_id: int
    year_id: int

# Properties to receive on subject creation
class SubjectCreate(SubjectBase):
    pass

# Properties to receive on subject update
class SubjectUpdate(SubjectBase):
    subject_name: Optional[str] = None
    subject_code: Optional[str] = None
    major_id: Optional[int] = None
    year_id: Optional[int] = None

# Properties shared by models stored in DB
class SubjectInDBBase(SubjectBase):
    subject_id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        orm_mode = True

# Properties to return to client
class Subject(SubjectInDBBase):
    pass

File: fe/university-docs/types/subject.ts
export interface Subject {
  subject_id: number
  subject_name: string
  subject_code: string
  description?: string
  major_id: number
  major_name: string
  year_id: number
  year_name: string
  created_at: string
  updated_at: string | null
}

export interface SubjectCreate {
  subject_name: string
  subject_code: string
  description?: string
  major_id: number
  year_id: number
}

export interface SubjectUpdate {
  subject_name?: string
  subject_code?: string
  description?: string
  major_id?: number
  year_id?: number
} 